bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -724,69 +730,63 @@ public Image(Device device, ImageFileNameProvider imageFileNameProvider) {
 public Image(Device device, ImageDataProvider imageDataProvider) {
 	super(device);
 	this.imageDataProvider = imageDataProvider;
-	currentNativeZoom = DPIUtil.getNativeDeviceZoom();
+	initialNativeZoom = DPIUtil.getNativeDeviceZoom();
 	ElementAtZoom<ImageData> data =  DPIUtil.validateAndGetImageDataAtZoom(imageDataProvider, getZoom());
-	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), data.zoom());
-	init (resizedData);
+	ImageData resizedData = DPIUtil.autoScaleImageData(device, data.element(), getZoom(), data.zoom());
+	init (resizedData, getZoom());
 	init();
 }
 
 /**
  * Update zoom and refresh the Image based on the native zoom level, if required.
  *
```

Comment:
* <b>IMPORTANT:</b> This method is <em>not</em> part of the public
 * API for <code>Image</code>. It is marked public only so that it
 * can be shared within the packages provided by SWT. It is not
 * available on all platforms, and should never be called from
 * application code.
 * </p>
 *
 * @param image the image to adapt for the provided zoom
 * @param targetZoom zoom in % of the standard resolution
 * @return an image object equal to the specified image scaled to the provided targetZoom

and

 * @noreference This method is not intended to be referenced by clients.
is missing in this comment

And remove the @since then